AssetRipper.Library/Exporters/Audio/AudioMixerExportCollection.cs

```diff
@@ -0,0 +1,272 @@
+				var snapshot = snapshotPPtr.GetAsset(mixer.SerializedFile) as IAudioMixerSnapshotController;
+				AddAsset(snapshot);
+				
+				var snapshotConstant = constants.Snapshots[i];
+				for (int j = 0; j < snapshotConstant.Values.Length; j++)
+				{
+					if (indexToGUID.TryGetValue((uint)j, out var valueGUID))
+					{
+						snapshot.FloatValues_C245[valueGUID] = snapshotConstant.Values[j];
+					}
+					else
+					{
+						Logger.Warning(LogCategory.Export, $"Snapshot({snapshot.Name_C245}) value #{j} has no binding parameter");
+					}
+				}
+
+				for (int j = 0; j < snapshotConstant.TransitionIndices.Length; j++)
+				{
+					uint paramIndex = snapshotConstant.TransitionIndices[j];
+					int transitionType = (int)snapshotConstant.TransitionTypes[j];
+					if (indexToGUID.TryGetValue(paramIndex, out var paramGUID))
+					{
+						snapshot.TransitionOverrides_C245[paramGUID] = transitionType;
+					}
+					else
+					{
+						Logger.Warning(LogCategory.Export, $"Snapshot({snapshot.Name_C245}) transition #{paramIndex} has no binding parameter");
+					}
+				}
+			}
+			
+			// generate exposed parameters
+
+			for (int i = 0; i < constants.ExposedParameterIndices.Length; i++)
+			{
+				uint paramIndex = constants.ExposedParameterIndices[i];
+				uint paramNameCrc = constants.ExposedParameterNames[i];
+				if (indexToGUID.TryGetValue(paramIndex, out var paramGUID))
+				{
+					var exposedParam = mixer.ExposedParameters_C241.AddNew();
+					exposedParam.Guid.CopyValues(paramGUID);
+					exposedParam.NameString = $"{CrcUtils.ReverseDigestAscii(paramNameCrc)}";
+				}
+				else
+				{
+					Logger.Warning(LogCategory.Export, $"Exposed parameter #{paramIndex} has no binding parameter");
+				}
+			}
+			
+			// complete mixer controller
+			
+			var groupView = mixer.AudioMixerGroupViews_C241.AddNew();
+			groupView.NameString = "View";
+			foreach (var group in groups)
+			{
+				groupView.Guids.Add(group.GroupID_C243);
+			}
+			mixer.CurrentViewIndex_C241 = 0;
+			mixer.TargetSnapshot_C241.CopyValues(mixer.StartSnapshot_C241);
+		}
+
+		private GUID IndexingNewGUID(uint index, Dictionary<uint, GUID> table)
+		{
+			var guid = (GUID)UnityGUID.NewGuid();
+			if (!table.TryAdd(index, guid))
+			{
+				Logger.Warning(LogCategory.Export, $"Constant index #{index} conflicts with another one.");
+			}
+			return guid;
+		}
+
+		private List<Utf8String> ParseNameBuffer(byte[] buffer)
+		{
+			var names = new List<Utf8String>();
+			int offset = 0;
+			while (buffer[offset] != 0)
+			{
+				int start = offset;
+				while (buffer[++offset] != 0) { }
+
+				var utf8Data = buffer.SubArray(start, offset - start);
+				names.Add(new Utf8String { Data = utf8Data });
+
+				offset++;
+			}
+
+			return names;
+		}
+
+		private uint m_nextExportID;
+		
+		protected override long GenerateExportID(IUnityObjectBase asset)
+		{
+			long exportID = ExportIdHandler.GetMainExportID(asset, m_nextExportID);
+			m_nextExportID += 2;
+			return exportID;
```

Comment:
Is this normal behavior, or do normal assets have random ids?

Reply:
This code didn't exist here on my first attempt. Then, when I tested against Unity 5.0, the editor crashed when opening the exported project. I found out this is related to the long exportIds.
Then I copied this code from TextureExportCollection, and everything is ok.

By the way, I have put this code into the base class in my "deterministic-export" branch. It works fine for me. You might consider it too.

Reply:
The crash issue doesn't exist for newer Unity versions, btw.

Reply:
Ok